clippy_lints/src/manual_slice_size_calculation.rs

```diff
@@ -80,13 +81,14 @@ fn simplify_half<'tcx>(
     cx: &LateContext<'tcx>,
     expr1: &'tcx Expr<'tcx>,
     expr2: &'tcx Expr<'tcx>,
-) -> Option<&'tcx Expr<'tcx>> {
+) -> Option<(&'tcx Expr<'tcx>, usize)> {
     if !expr1.span.from_expansion()
         // expr1 is `[T1].len()`?
         && let ExprKind::MethodCall(method_path, receiver, _, _) = expr1.kind
         && method_path.ident.name == sym::len
         && let receiver_ty = cx.typeck_results().expr_ty(receiver)
-        && let ty::Slice(ty1) = receiver_ty.peel_refs().kind()
+        && let (receiver_ty, refs_count) = peel_refs_and_count(receiver_ty)
```

Comment:
You should use `clippy_utils::ty::walk_ptrs_ty_depth()` from ` instead.

Reply:
Thanks for your kind review, and I have changed this line applying your suggestion. :heart: